pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1982,14 +1985,18 @@ func (p *Provider) CreateLoadBalancer(_ *logger.Logger, vms vm.List, port int) e
 		return errors.Errorf("no managed instance groups found for cluster %s", clusterName)
 	}
 
+	var args []string
 	healthCheckName := loadBalancerResourceName(clusterName, port, "health-check")
-	args := []string{"compute", "health-checks", "create", "tcp",
-		healthCheckName,
-		"--project", project,
-		"--port", strconv.Itoa(port),
-	}
-	cmd := exec.Command("gcloud", args...)
-	output, err := cmd.CombinedOutput()
+	output, err := func() ([]byte, error) {
```

Comment:
Would adding something like:

```
func (s *Spinner) Run(f func()) {
	defer s.Start()()
	f()
}

----------

var output []byte
ui.NewDefaultSpinner(l, "created health check").Run(func() {
    output, err = cmd.CombinedOutput()
})
```

be slightly cleaner? Although in this you have to declare `output` so I'm not actually sure it's cleaner/less boilerplate.

Reply:
Yeah I also thought a `Run` function could be a nice to have. Not opposed to adding one, the pain I imagine would be error handling on returns etc would have to be outside.

At this moment I wished there was some way to have a "generic" return pass-through for the `Run`. Alternatively may be good just to provide an error pass-through/wrapper variant of `Run`, then it would just be a bit more nuanced if you needed to return some output as well, which you'll have to capture outside the lambda.